joomla/joomla-cms
# [4.2] Template Manager: Add SCSS compiler

### Summary of Changes
This PR adds a SCSS compiler to the template manager to compile SCSS files from Joomla directly without using Node or similar stuff. Basically, this is the compiler we had in Joomla 3, but with SCSS instead of LESS. As compiler it uses https://github.com/scssphp/scssphp, which seems to be rather active. Right now this does compile the template.scss from Cassiopeia without issues.

Right now this is a draft for several reasons, which are listed below.

### Testing Instructions
1. Apply patch and go to the template manager.
2. Edit the files of a template and edit a SCSS file, for example in the scss folder of Cassiopeia.
3. See a new "Compile SCSS" button at the top and click it.
4. It takes quite some time, but see that the SCSS has been compiled to CSS including a SourceMap.

### Issues/Todos
1. Right now the compiler outputs only a non-minified version of the CSS, as well as a SourceMap. Ideally it should create a minified and non-minified version of the CSS and a SourceMap for both.
2. I think right now this will only compile SCSS files in the `/scss/` folder of the template to the `/css/` folder, but not from subfolders.
3. The template manager should have options for minifying the CSS and creating SourceMaps for these files.
4. Ideally, we would allow to compile this on the fly. But for that we are lacking the API right now.
5. The SourceMap, which is currently created, contains the full path to the SCSS files, which is a security issue.

I'm happy for others to support this PR and to provide improvements.
